src/platforms/stlink/stlink_common.c

```diff
@@ -45,6 +48,17 @@ uint32_t detect_rev(void)
 	rcc_periph_reset_pulse(RST_USB);
 	rcc_periph_clock_enable(RCC_AFIO);
 	rcc_periph_clock_enable(RCC_CRC);
+
+#if defined(STLINK_FORCE_CLONE)
+	/* PA12 is USB D+ pin */
+	gpio_clear(GPIOA, GPIO12);
+	gpio_set_mode(GPIOA, GPIO_MODE_OUTPUT_2_MHZ, GPIO_CNF_OUTPUT_OPENDRAIN, GPIO12);
```

Comment:
This appears to be new pin setup code for clone models - could you explain briefly why this is required and correct?

Reply:
It is not new, it was executed before at the end of the function(the if statement based on rev) and talking to @ALTracer it is used to force USB re enumeration after reset from the bootloader. I decided to delete the code which is used for rev detection and moved these 2 lines there

Reply:
Ohhh, gotcha - ok, thank you for clarifying!

Reply:
Indeed I gathered that this is a last-resort hack to present a long SE0 state on fixed 1.5k pullup boards to force re-enumeration. By the way, how do you expect to do it? Is PA15 USB_RENUM wired and available? Or this platform also has to perform the hack? (of inducing a babble (?) condition for lack of togglable pullup voltage source pin per USB 1.1/2.0 spec) Please confirm that the board works as BMF with ST_BOOTLOADER=1, unless you already wiped your stock bootloader with BMPBootloader/F1 (it may have the toggling because it calls `detect_rev()`)